webui/react/src/e2e/utils/naming.ts

```diff
@@ -19,3 +9,28 @@ export const randIdAlphanumeric = (): string => Math.random().toString(36).subst
  * @returns Numeric hash
  */
 export const randId = (): number => Math.floor(Math.random() * 10_000);
+
+/**
+ * Generates a naming function
+ * @returns The a naming function which uses the same session hash across any imports.
+ */
+function genSafeName(): { sessionRandomHash: string; safeName: (baseName: string) => string } {
+  // include _ in the hash to avoid naming collisions.
+  // for example, we would want to avoid searching for the name "admin", and _ helps with that.
+  const sessionRandomHash = `_${randIdAlphanumeric()}`;
+  /**
+   * Appends a timestamp and a short random hash to a base name to avoid naming collisions.
+   * @param baseName The base name to append the timestamp and hash to.
+   * @returns The base name with appended timestamp and hash.
+   */
+  return {
+    safeName: (baseName: string): string => {
+      const timestamp = Date.now();
+      return `${baseName}${sessionRandomHash}${randIdAlphanumeric()}_${timestamp}`;
```

Comment:
> meaningful information in unique IDs
that's a good point. i always think it's best to name a test case rather than generate it's name from the types of parameters it has, for example.

> Could we just use a simpler UUID
the session random hash is specifically useful in situations like this one. using the session hash, I can find all the users i made "during this test session" with
```
await userManagementPage.search.pwLocator.fill(usernamePrefix + sessionRandomHash);
```
rather than all the users that just match the prefix, which would collide.
<img width="493" alt="image" src="https://github.com/determined-ai/determined/assets/5224116/b658e82d-d920-4d21-a1a9-fae2b6a2853c">
if i had 10 unique ID users, there's no way I could gather them all with a single search unless I used a prefix, but that prefix would be guaranteed to fail on a dirty environment. After the hashes, I figured it would be useful to anyone using a dirty environment to see a timestamp. I agree that we shouldn't be relying on the timestamp for collision or test meaning.

if it's alright with you, I'll throw this explanation into a docstring or readme-- I really need to keep the session hash. I don't know another way around it. Also, it's not `2n` calls to `Math.random()`, it's `n+1`.